src/components/style-selector/medium-style-item.vue

```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+import { useTranslation } from 'i18next-vue'
+import { StyleItem } from '@/composables/mvt-styles/mvt-styles.model'
+
+const { t } = useTranslation()
+const props = defineProps<{
+  style: StyleItem
+  colorEditable: boolean
+}>()
+const emit = defineEmits<{
+  (e: 'changeStyle', style: StyleItem): void
+}>()
+
+function updateColor(colorChangeEvent: Event) {
+  const newStyle = Object.assign(Object.assign({}, props.style), {
+    color: colorChangeEvent.target.value,
+  })
```

Comment:
Use spread operator ?
```suggestion
const newStyle = {
  ...props.style,
  color: colorChangeEvent.target.value,
};
```

Reply:
good idea, thanks